### DISCUSSION TITLE
Renovate doesn't assign correct Reviewer, Assignee and doesn't set MR status to Ready when matching packageRules

### DISCUSSION THREAD

**Discussion by liby** [renovatebot/renovate#22614] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.113.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Hello,

I'm experiencing a peculiar issue with Renovate where it doesn't assign the correct Reviewer and Assignee, and also doesn't set the status of the Merge Request (MR) to Ready when the MR matches certain packageRules in my configuration.

Here's the related packageRules section from my config:
```
  "packageRules": [
    {
      "groupName": "Dependencies (non-major)",
      "groupSlug": "all-minor-patch-pin",
      "matchUpdateTypes": ["patch", "minor", "pin"]
    }
  ]
```

And the top-level reviewers setting:

```
{
  "reviewers": ["namex"]
}
```

In the case of MRs that don't match any `packageRules`, the correct Reviewer ("namex") is assigned, the Assignee is set correctly (not to the bot itself), and the status of the MR is set to Ready as expected. However, for MRs that match the aforementioned package rule, the bot assigns itself as the Assignee, doesn't assign any reviewer, and the MR status isn't set to Ready.

I've reviewed my configuration and couldn't find any apparent mistakes that would lead to this behavior. Also, there were no relevant errors or warnings in the Renovate logs.

I suspect this could be a bug with Renovate's handling of packageRules. Could you please assist in identifying if there's a problem with my configuration or if it's an issue with Renovate?
<img width="1517" alt="Xnip2023-06-07_10-15-06" src="https://github.com/renovatebot/renovate/assets/38807139/db67c53d-9319-4310-8762-f586eb86127b">
<img width="1427" alt="Xnip2023-06-07_10-22-50" src="https://github.com/renovatebot/renovate/assets/38807139/1c775141-eca0-4e73-9b4a-b27e668f6d2b">

### Relevant debug logs

[renovate.log](https://github.com/renovatebot/renovate/files/11672040/renovate.log)

### Have you created a minimal reproduction repository?

n/a

**Comment by rarkins** [comment]
You have another problem which should be addressed first.

```
DEBUG: Pull request creation error (repository=7, branch=feature/renovate-all-minor-patch-pin)
       "err": {
         "name": "HTTPError",
         "code": "ERR_NON_2XX_3XX_RESPONSE",
         "timings": {
           "start": 1686098656698,
           "socket": 1686098656698,
           "lookup": 1686098656703,
           "connect": 1686098656703,
           "secureConnect": 1686098656708,
           "upload": 1686098656709,
           "response": 1686098656832,
           "end": 1686098656832,
           "phases": {
             "wait": 0,
             "dns": 5,
             "tcp": 0,
             "tls": 5,
             "request": 1,
             "firstByte": 123,
             "download": 0,
             "total": 134
           }
         },
         "message": "Response code 409 (Conflict)",
         "stack": "HTTPError: Response code 409 (Conflict)\n    at Request.<anonymous> (/opt/containerbase/tools/renovate/35.113.0/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/35.113.0 (https://github.com/renovatebot/renovate)",
             "accept": "application/json",
             "authorization": "***********",
             "content-type": "application/json",
             "content-length": "22306",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://gitlab.company.io/api/v4/projects/7/merge_requests",
           "hostType": "gitlab",
           "username": "",
           "password": "",
           "method": "POST",
           "http2": false
         },
         "response": {
           "statusCode": 409,
           "statusMessage": "Conflict",
           "body": {
             "message": [
               "Another open merge request already exists for this source branch: !4051"
             ]
           },
           "headers": {
             "date": "Wed, 07 Jun 2023 00:44:16 GMT",
             "content-type": "application/json",
             "content-length": "87",
             "connection": "close",
             "cache-control": "no-cache",
             "vary": "Origin",
             "x-content-type-options": "nosniff",
             "x-frame-options": "SAMEORIGIN",
             "x-gitlab-meta": "{\"correlation_id\":\"01H29MZFE69AG1CR6WTG1FW5DA\",\"version\":\"1\"}",
             "x-request-id": "01H29MZFE69AG1CR6WTG1FW5DA",
             "x-runtime": "0.119335",
             "strict-transport-security": "max-age=63072000",
             "referrer-policy": "strict-origin-when-cross-origin"
           },
           "httpVersion": "1.1",
           "retryCount": 0
         }
       }
```

When Renovate went to create the MR, it found that another MR already existed. This is usually because of a mismatch in `username`.

You should also keep in mind that Renovate sets assignees and reviewers _at the time of MR creation_ only. So the entire log you gave is not useful because the MR already existed, and reviewers/assignees were decided previously

**Reply by liby** [reply]
Sorry, you're absolutely right.

I just searched through all the pipelines, and I finally found the log from when this Merge Request was created.

[auto-merge-non-major-updates.log](https://github.com/renovatebot/renovate/files/11673762/auto-merge-non-major-updates.log)

**Reply by rarkins** [reply]
I only find this:

```
DEBUG: Reached branch limit - skipping branch creation (repository=7, branch=feature/renovate-all-minor-patch-pin)
```

**Reply by liby** [reply]
But it was indeed accidentally created, and all subsequent logs say that this branch already exists.

Moreover, there are no corresponding settings to limit the number of PRs.
